File: app/src/test/java/com/vereshchagin/nikolay/stankinschedule/PairTest.java
package com.vereshchagin.nikolay.stankinschedule;

import com.google.gson.GsonBuilder;
import com.vereshchagin.nikolay.stankinschedule.schedule.model.pair.Pair;

import org.apache.commons.io.FileUtils;
import org.junit.Test;

import java.io.File;
import java.io.IOException;
import java.nio.charset.StandardCharsets;

import static org.junit.Assert.assertTrue;

/**
 * Тесты связанные с парой.
 */
public class PairTest {

    private static final String PATH = "src/test/resources/";

    /**
     * Проверка на правильность загрузки/сохранения пары.
     */
    @Test
    public void loadingAndSaving() {
        try {
            for (int i = 1; i <= 6; i++) {
                File file = FileUtils.getFile(PATH, String.format("pair_%d.json", i));
                String json = FileUtils.readFileToString(file, StandardCharsets.UTF_8);

                Pair pair = new GsonBuilder()
                        .registerTypeAdapter(Pair.class, new Pair.PairDeserialize())
                        .create()
                        .fromJson(json, Pair.class);

                new GsonBuilder()
                        .registerTypeAdapter(Pair.class, new Pair.PairSerialize())
                        .setPrettyPrinting()
                        .create()
                        .toJson(pair);
            }

            assertTrue("Successfully!", true);

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/test/java/com/vereshchagin/nikolay/stankinschedule/PairTest.kt
package com.vereshchagin.nikolay.stankinschedule

import com.google.gson.GsonBuilder
import com.vereshchagin.nikolay.stankinschedule.resources.PairResources
import org.junit.Assert
import org.junit.Test

/**
 * Тесты для пары в расписании.
 */
class PairTest {

    private val gson = GsonBuilder()
        .setPrettyPrinting()
        .create()

    /**
     * Проверка на правильность загрузки/сохранения пары.
     */
    @Test
    fun loadingAndSavingPairs() {
        val pairs = PairResources.PAIRS.map { json ->
            gson.fromJson(json, Pair::class.java)
        }

        val newPairs = PairResources.PAIRS.map { json ->
            val pair = gson.fromJson(json, Pair::class.java)
            val newJson = gson.toJson(pair)
            gson.fromJson(newJson, Pair::class.java)
        }

        Assert.assertEquals(pairs, newPairs)
    }

    /**
     * Проверка на сравнение пар.
     */
    @Test
    fun pairEquals() {
        val first = PairResources.PAIRS.map { json ->
            gson.fromJson(json, Pair::class.java)
        }
        val second = PairResources.PAIRS.map { json ->
            gson.fromJson(json, Pair::class.java)
        }

        Assert.assertEquals(first, second)
    }
}